skt/runner.py

```diff
@@ -325,12 +325,17 @@ def getresults(self, jobid=None):
 
     def recipe_to_job(self, recipe, samehost=False):
         tmp = recipe.copy()
+
+        hreq = tmp.find("hostRequires")
         if samehost:
-            hreq = tmp.find("hostRequires")
             hostname = etree.Element("hostname")
             hostname.set("op", "=")
             hostname.set("value", tmp.attrib.get("system"))
             hreq.append(hostname)
+        else:
+            hostname = hreq.find('hostname')
+            if hostname is not None:
+                hreq.remove(hostname)
```

Comment:
Hmm, shouldn't we simply remove the `hostname` requirement every time, and only add it back when `samehost` is True?

Reply:
I don't understand how you mean it. If the `samehost` is true, we keep the hostname and if not, we check if there is a hostname specified in the recipe and remove it if there is one. Can you elaborate how you imagine the solution?

Reply:
Like this:

        hreq = tmp.find("hostRequires")
        hostname = hreq.find('hostname')
        if hostname is not None:
            hreq.remove(hostname)
        if samehost:
            hostname = etree.Element("hostname")
            hostname.set("op", "=")
            hostname.set("value", tmp.attrib.get("system"))
            hreq.append(hostname)

This way we make sure there's no `hostname` element when `samehost` is `False`, and there's only one `hostname` element, when it's `True` (that is unless there was more than one originally, which we can also handle, if needed).

Reply:
Thanks for the explanation! Yes, that makes sense. I'll update the pull.